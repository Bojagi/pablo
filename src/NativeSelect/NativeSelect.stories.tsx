import React from 'react';

import { Search } from 'react-feather';
import { css } from 'styled-components';
import { Box } from '../Box';
import { IconButton } from '../IconButton';
import { Typography } from '../Typography';
import { NativeSelect, NativeSelectProps } from './NativeSelect';

export default {
  title: 'NativeSelect',
};

const ControlledSelect = ({
  value: valueInitial,
  ...props
}: Omit<NativeSelectProps, 'onChange'>) => {
  const [value, setValue] = React.useState(valueInitial);
  return <NativeSelect value={value} mb={4} {...props} onChange={setValue} />;
};

const baseStory = (args) => (
  <ControlledSelect {...args}>
    <option>Hans</option>
    <option>Ilse</option>
    <option>Franz</option>
  </ControlledSelect>
);

export const FullWidth = baseStory.bind(null);
FullWidth.args = { fullWidth: true };

export const FixedWidth = baseStory.bind(null);
FixedWidth.args = { width: 800 };

export const Inline = baseStory.bind(null);
Inline.args = {};

export const WithLabel = baseStory.bind(null);
WithLabel.args = {
  label: <>name</>,
};

export const WithInfoText = baseStory.bind(null);
WithInfoText.args = {
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const WithLabelAndInfoText = baseStory.bind(null);
WithLabelAndInfoText.args = {
  label: <>name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const Error = baseStory.bind(null);
Error.args = {
  error: 'Something terrible happened!',
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
};

export const MultipleInputs = () => (
  <Box>
    <Box mb={5}>
      <ControlledSelect label="first name">
        <option>Hans</option>
        <option>Ilse</option>
        <option>Franz</option>
      </ControlledSelect>
    </Box>
    <Box mb={5}>
      <ControlledSelect label="last name">
        <option>Wurst</option>
        <option>Pilse</option>
        <option>Kranz</option>
      </ControlledSelect>
    </Box>
    <Box mb={5}>
      <ControlledSelect value="123456789" label="credit card no." infoText="optional">
        <option>None</option>
        <option>123456789</option>
      </ControlledSelect>
    </Box>
  </Box>
);

export const WithEndComponent = baseStory.bind(null);
WithEndComponent.args = {
  label: <>search</>,
  fullWidth: false,
  width: 600,
  end: (
    <IconButton size="small">
      <Search />
    </IconButton>
  ),
};

export const WithStartComponent = baseStory.bind(null);
WithStartComponent.args = {
  fullWidth: false,
  width: 600,
  start: (
    <Typography mb={0} variant="paragraphBold">
      search:
    </Typography>
  ),
};

export const Outline = baseStory.bind(null);
Outline.args = { variant: 'outline' };

export const WithCustomStyles = baseStory.bind(null);
WithCustomStyles.args = {
  label: <>Some longer name</>,
  infoText: (
    <>
      This is something <strong>important!</strong>
    </>
  ),
  customStyles: {
    label: css`
      transform: translateY(-36px) rotate(6deg);
      /* transform-origin: 0 0; */
      text-align: right;
    `,
    infoText: css`
      transform: rotate(-3deg);
      transform-origin: 0 0;
    `,
    wrapper: css`
      transform: rotate(-5deg);
      transform-origin: 0 0;
      background-color: blue;
    `,
    field: css`
      color: red;
    `,
  },
};
