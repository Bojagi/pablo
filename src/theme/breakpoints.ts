export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
