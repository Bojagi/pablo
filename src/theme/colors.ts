export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
