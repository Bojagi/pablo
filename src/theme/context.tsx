import * as React from 'react';
import { PabloTheme, ComponentStyles } from './types';

export const pabloThemeContext = React.createContext<PabloTheme>({} as any);
export const pabloComponentStylesContext = React.createContext<ComponentStyles>({} as any);
export const rootContext = React.createContext<Document | ShadowRoot | null>(null);

export const useComponentStyleContext = () => React.useContext(pabloComponentStylesContext);

export const usePabloTheme = () => React.useContext(pabloThemeContext);
