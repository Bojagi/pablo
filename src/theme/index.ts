export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
