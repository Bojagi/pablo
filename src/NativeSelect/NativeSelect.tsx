import React, { forwardRef, ReactElement } from 'react';
import styled from '@emotion/styled';
import { LayoutBoxProps } from '../Box';
import { getComponentStyle } from '../styleHelpers/getComponentStyle';
import { BaseInput, InnerInputProps, InputVariant } from '../shared/BaseInput';
import { useComponentStyle } from '../theme/useComponentStyle';
import { getCustomStyles } from '../utils/useCustomStyles';
import { baseStyle } from '../shared/baseStyle';

export interface NativeSelectProps extends LayoutBoxProps {
  id?: string;
  value?: string | number;
  inputRef?: React.Ref<HTMLSelectElement>;
  children: ReactElement<'option'>[];
  error?: React.ReactNode;
  label?: React.ReactNode;
  variant?: InputVariant;
  infoText?: React.ReactNode;
  fullWidth?: boolean;
  end?: React.ReactNode;
  onChange: (newValue: string, e: React.FormEvent<HTMLSelectElement>) => void;
}

const StyledSelect = styled.select<InnerInputProps>`
  ${baseStyle}
  flex-grow: 1;
  border: 0;
  appearance: none;
  padding: ${getComponentStyle('nativeSelect.padding')};
  background-color: transparent;
  font-family: ${getComponentStyle('nativeSelect.fontFamily')};
  outline: none;
  padding-right: ${getComponentStyle('nativeSelect.reservedArrowSpace')};
  position: relative;
  ${getCustomStyles('nativeSelect.styles', 'field')}
`;

const SelectWrapper = styled.div`
  position: relative;
  width: 100%;
  display: flex;

  :after {
    content: '';
    background-image: url("data:image/svg+xml,%3Csvg width='8' height='6' viewBox='0 0 8 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M4 6L0.535899 -6.52533e-07L7.4641 -4.68497e-08L4 6Z' fill='currentColor' /%3E%3C/svg%3E");
    width: 8px;
    height: 6px;
    position: absolute;
    right: ${getComponentStyle('nativeSelect.arrowGap')};
    top: 50%;
    transform: translateY(-50%);
  }
`;

const InnerSelect = forwardRef<HTMLSelectElement, React.ComponentProps<typeof StyledSelect>>(
  (props, ref) => (
    <SelectWrapper>
      <StyledSelect ref={ref} {...props} />
    </SelectWrapper>
  )
) as unknown as typeof StyledSelect;

export const NativeSelect = forwardRef<HTMLDivElement, NativeSelectProps>(
  ({ width, variant = 'filled', ...props }, ref) => {
    const defaultWidth = useComponentStyle('nativeSelect.defaultWidth') as any;
    const adornmentGap = useComponentStyle('nativeSelect.adornmentGap');
    return (
      <BaseInput<InnerInputProps, any>
        name="nativeSelect"
        innerRef={ref}
        variant={variant}
        adornmentGap={adornmentGap}
        inputComponent={InnerSelect}
        {...props}
        width={width || defaultWidth}
      />
    );
  }
);
