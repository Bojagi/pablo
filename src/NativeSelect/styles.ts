import { css } from 'styled-components';
import { inputBaseStyles, InputBaseStyles } from '../shared/inputBaseStyles';
import { Style } from '../theme/types';
import { getSpacing } from '../styleHelpers';

export interface NativeSelectStyles extends InputBaseStyles {
  adornmentGap: Style;
  arrowGap: Style;
  reservedArrowSpace: Style;
}

export const nativeSelectStyles: NativeSelectStyles = {
  ...inputBaseStyles,
  adornmentGap: getSpacing(4),
  arrowGap: getSpacing(4),
  reservedArrowSpace: css`calc(${getSpacing(4)} * 2 + 8px)`,
};
