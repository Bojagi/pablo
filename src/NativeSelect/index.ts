import './styles';

export * from './NativeSelect';
