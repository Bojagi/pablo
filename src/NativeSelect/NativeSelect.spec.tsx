import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { NativeSelect } from './NativeSelect';

test('Render without value', () => {
  const { container, getByTestId } = renderComponent({});
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-nativeSelect')).not.toHaveValue();
});

test('Render with value', () => {
  const { getByTestId } = renderComponent({ value: 'some' });
  expect(getByTestId('pbl-nativeSelect')).toHaveValue('some');
});

test('Show label when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    label: (
      <>
        Your <strong>name</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-nativeSelect-label')).toHaveTextContent('Your name');
});

test('Show infotext when available', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    infoText: (
      <>
        Something that <strong>helps</strong>
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-nativeSelect-infotext')).toHaveTextContent('Something that helps');
});

test('Show error when an error message is given', () => {
  const { container, getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
  });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-nativeSelect-infotext')).toHaveTextContent(
    'Something terrible happened!'
  );
});

test('Show error instead of infotext when both are set', () => {
  const { getByTestId } = renderComponent({
    value: 'some',
    error: (
      <>
        Something <strong>terrible</strong> happened!
      </>
    ),
    infoText: (
      <>
        something that <strong>helps</strong>
      </>
    ),
  });
  expect(getByTestId('pbl-nativeSelect-infotext')).toHaveTextContent(
    'Something terrible happened!'
  );
});

test('Render with custom width when "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: '600em',
  });

  expect(getByTestId('pbl-nativeSelect-wrapper')).toHaveStyleRule('width', '600em');
});

test('Render with custom px width when number "width" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
  });

  expect(getByTestId('pbl-nativeSelect-wrapper')).toHaveStyleRule('width', '600px');
});

test('Render with width auto when "fullWidth" prop is set', () => {
  const { getByTestId } = renderComponent({
    width: 600,
    fullWidth: true,
  });

  expect(getByTestId('pbl-nativeSelect-wrapper')).toHaveStyleRule('width', 'auto');
});

test('Trigger onChange event when the content of the inner input changed', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    onChange: onChangeMock,
  });

  expect(onChangeMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.change(getByTestId('pbl-nativeSelect'), { target: { value: 'otherValue' } });
  });
  expect(onChangeMock).toHaveBeenCalledTimes(1);
  expect(onChangeMock).toHaveBeenCalledWith(
    'otherValue',
    expect.objectContaining({ type: 'change' })
  );
});

test('Use generated id when no "id" prop is set', () => {
  const onChangeMock = jest.fn();
  const { getByTestId } = renderComponent({
    value: 'some',
    id: undefined,
    onChange: onChangeMock,
  });

  expect(getByTestId('pbl-nativeSelect')).toHaveAttribute(
    'id',
    expect.stringContaining('pbl-nativeSelect-')
  );
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <NativeSelect id="custom-id" {...props}>
        <option value="">---</option>
        <option value="some">Some</option>
        <option value="otherValue">Other</option>
      </NativeSelect>
    </PabloThemeProvider>
  );
}
