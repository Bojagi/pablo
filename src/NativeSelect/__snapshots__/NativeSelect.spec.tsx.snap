// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without value 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid var(--pbl-theme-colors-borders-main);
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-common-white);
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.emotion-2 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2:after {
  content: '';
  background-image: url("data:image/svg+xml,%3Csvg width='8' height='6' viewBox='0 0 8 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M4 6L0.535899 -6.52533e-07L7.4641 -4.68497e-08L4 6Z' fill='currentColor' /%3E%3C/svg%3E");
  width: 8px;
  height: 6px;
  position: absolute;
  right: 0.5rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0.375rem 0.5rem;
  background-color: transparent;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  outline: none;
  padding-right: calc(0.5rem * 2 + 8px);
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="pbl-nativeSelect-wrapper"
      name="nativeSelect"
      width="300"
    >
      <div
        class="emotion-2"
      >
        <select
          aria-invalid="false"
          class="emotion-3"
          data-testid="pbl-nativeSelect"
          id="custom-id"
        >
          <option
            value=""
          >
            ---
          </option>
          <option
            value="some"
          >
            Some
          </option>
          <option
            value="otherValue"
          >
            Other
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;

exports[`Show error when an error message is given 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid var(--pbl-theme-colors-borders-main);
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-common-white);
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  border-color: var(--pbl-theme-colors-negative-main);
  width: 300px;
}

.emotion-1:focus {
  box-shadow: 0 0 0 3px var(--pbl-theme-colors-negative-lightest);
}

.emotion-2 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2:after {
  content: '';
  background-image: url("data:image/svg+xml,%3Csvg width='8' height='6' viewBox='0 0 8 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M4 6L0.535899 -6.52533e-07L7.4641 -4.68497e-08L4 6Z' fill='currentColor' /%3E%3C/svg%3E");
  width: 8px;
  height: 6px;
  position: absolute;
  right: 0.5rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0.375rem 0.5rem;
  background-color: transparent;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  outline: none;
  padding-right: calc(0.5rem * 2 + 8px);
  position: relative;
}

.emotion-4 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-info-lineHeight);
  font-size: var(--pbl-theme-typography-info-fontSize);
  margin-bottom: var(--pbl-theme-typography-info-marginBottom);
  margin-top: 0.25rem;
  color: var(--pbl-theme-colors-negative-main);
  margin-top: 0.25rem;
  color: var(--pbl-theme-colors-negative-main);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="pbl-nativeSelect-wrapper"
      name="nativeSelect"
      width="300"
    >
      <div
        class="emotion-2"
      >
        <select
          aria-errormessage="custom-id-error"
          aria-invalid="true"
          class="emotion-3"
          data-testid="pbl-nativeSelect"
          id="custom-id"
        >
          <option
            value=""
          >
            ---
          </option>
          <option
            value="some"
          >
            Some
          </option>
          <option
            value="otherValue"
          >
            Other
          </option>
        </select>
      </div>
    </div>
    <p
      aria-live="polite"
      class="emotion-4"
      data-testid="pbl-nativeSelect-infotext"
      id="custom-id-error"
    >
      Something 
      <strong>
        terrible
      </strong>
       happened!
    </p>
  </div>
</div>
`;

exports[`Show infotext when available 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid var(--pbl-theme-colors-borders-main);
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-common-white);
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.emotion-2 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2:after {
  content: '';
  background-image: url("data:image/svg+xml,%3Csvg width='8' height='6' viewBox='0 0 8 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M4 6L0.535899 -6.52533e-07L7.4641 -4.68497e-08L4 6Z' fill='currentColor' /%3E%3C/svg%3E");
  width: 8px;
  height: 6px;
  position: absolute;
  right: 0.5rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0.375rem 0.5rem;
  background-color: transparent;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  outline: none;
  padding-right: calc(0.5rem * 2 + 8px);
  position: relative;
}

.emotion-4 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-info-lineHeight);
  font-size: var(--pbl-theme-typography-info-fontSize);
  margin-bottom: var(--pbl-theme-typography-info-marginBottom);
  margin-top: 0.25rem;
  color: var(--pbl-theme-colors-text-info);
  margin-top: 0.25rem;
  color: var(--pbl-theme-colors-text-info);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="pbl-nativeSelect-wrapper"
      name="nativeSelect"
      width="300"
    >
      <div
        class="emotion-2"
      >
        <select
          aria-describedby="custom-id-info"
          aria-invalid="false"
          class="emotion-3"
          data-testid="pbl-nativeSelect"
          id="custom-id"
        >
          <option
            value=""
          >
            ---
          </option>
          <option
            value="some"
          >
            Some
          </option>
          <option
            value="otherValue"
          >
            Other
          </option>
        </select>
      </div>
    </div>
    <p
      aria-live="polite"
      class="emotion-4"
      data-testid="pbl-nativeSelect-infotext"
      id="custom-id-info"
    >
      Something that 
      <strong>
        helps
      </strong>
    </p>
  </div>
</div>
`;

exports[`Show label when available 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

.emotion-1 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-paragraphBold-lineHeight);
  font-size: var(--pbl-theme-typography-paragraphBold-fontSize);
  margin-bottom: var(--pbl-theme-typography-paragraphBold-marginBottom);
  font-weight: var(--pbl-theme-typography-paragraphBold-fontWeight);
  margin-bottom: 0.375rem;
  margin-bottom: 0.375rem;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid var(--pbl-theme-colors-borders-main);
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-common-white);
  -webkit-transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: border-color 0.3s,box-shadow 0.3s ease-in-out;
  width: 300px;
}

.emotion-3 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3:after {
  content: '';
  background-image: url("data:image/svg+xml,%3Csvg width='8' height='6' viewBox='0 0 8 6' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M4 6L0.535899 -6.52533e-07L7.4641 -4.68497e-08L4 6Z' fill='currentColor' /%3E%3C/svg%3E");
  width: 8px;
  height: 6px;
  position: absolute;
  right: 0.5rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-4 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  padding: 0.375rem 0.5rem;
  background-color: transparent;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  outline: none;
  padding-right: calc(0.5rem * 2 + 8px);
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    <label
      data-testid="pbl-nativeSelect-label"
      for="custom-id"
    >
      <p
        class="emotion-1"
      >
        Your 
        <strong>
          name
        </strong>
      </p>
    </label>
    <div
      class="emotion-2"
      data-testid="pbl-nativeSelect-wrapper"
      name="nativeSelect"
      width="300"
    >
      <div
        class="emotion-3"
      >
        <select
          aria-invalid="false"
          class="emotion-4"
          data-testid="pbl-nativeSelect"
          id="custom-id"
        >
          <option
            value=""
          >
            ---
          </option>
          <option
            value="some"
          >
            Some
          </option>
          <option
            value="otherValue"
          >
            Other
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
